docker/images/dev-cli/Dockerfile

```diff
@@ -0,0 +1,59 @@
+# Creates an image for skycoin development
+FROM golang:1.10.0-stretch
+
+# Installs software
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        cmake \
+        libpcre3-dev \
+        gdbserver \
+        gdb \
+        vim \
+        ctags \
+        vim-scripts \
+        screen \
+        sudo \
+        doxygen \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* 
+
```

Comment:
@stdevGeekmidget at this point you gotta install `moxygen` . It requires `npm` so you might need to merge in rules found in `Dockerfile`'s for

- ~[buildpack-deps:stretch-curl](https://github.com/docker-library/buildpack-deps/blob/master/stretch/curl/Dockerfile)~ (not needed)

```
# dpkg -l ca-certificates curl netba
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  ca-certificate 20161130+nmu all          Common CA certificates
ii  curl           7.52.1-5+deb amd64        command line tool for transferrin
ii  dirmngr        2.1.18-8~deb amd64        GNU privacy guard - network certi
ii  gnupg          2.1.18-8~deb amd64        GNU privacy guard - a free PGP re
ii  netbase        5.4          all          Basic TCP/IP networking system
ii  wget           1.18-5+deb9u amd64        retrieves files from the web
```

- ~[buildpack-deps:buildpack-deps:stretch-scm](https://github.com/docker-library/buildpack-deps/blob/master/stretch/scm/Dockerfile)~ (not needed)

```
# dpkg -l git bzr mercurial subversion

Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  bzr            2.7.0+bzr661 all          easy to use distributed version c
ii  git            1:2.11.0-3+d amd64        fast, scalable, distributed revis
ii  mercurial      4.0-1+deb9u1 amd64        easy-to-use, scalable distributed
ii  subversion     1.9.5-1+deb9 amd64        Advanced version control system
```

- [buildpack-deps:stretch](https://github.com/docker-library/buildpack-deps/blob/master/stretch/Dockerfile)

```
# dpkg -l autoconf \
>         automake \
>         bzip2 \
>         dpkg-dev \
>         file \
>         g++ \
>         gcc \
>         imagemagick \
>         libbz2-dev \
>         libc6-dev \
>         libcurl4-openssl-dev \
>         libdb-dev \
>         libevent-dev \
>         libffi-dev \
>         libgdbm-dev \
>         libgeoip-dev \
>         libglib2.0-dev \
>         libjpeg-dev \
>         libkrb5-dev \
>         liblzma-dev \
>         libmagickcore-dev \
>         libmagickwand-dev \
>         libncurses5-dev \
>         libncursesw5-dev \
>         libpng-dev \
>         libpq-dev \
>         libreadline-dev \
>         libsqlite3-dev \
>         libssl-dev \
>         libtool \
>         libwebp-dev \
>         libxml2-dev \
>         libxslt-dev \
>         libyaml-dev \
>         make \
>         patch \
>         xz-utils \
> zlib1g-dev
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
un  autoconf       <none>       <none>       (no description available)
un  automake       <none>       <none>       (no description available)
un  bzip2          <none>       <none>       (no description available)
un  dpkg-dev       <none>       <none>       (no description available)
un  file           <none>       <none>       (no description available)
ii  g++            4:6.3.0-4    amd64        GNU C++ compiler
ii  gcc            4:6.3.0-4    amd64        GNU C compiler
ii  libc6-dev:amd6 2.24-11+deb9 amd64        GNU C Library: Development Librar
un  libtool        <none>       <none>       (no description available)
ii  make           4.1-9.1      amd64        utility for directing compilation
un  patch          <none>       <none>       (no description available)
un  xz-utils       <none>       <none>       (no description available)
dpkg-query: no packages found matching imagemagick
dpkg-query: no packages found matching libbz2-dev
dpkg-query: no packages found matching libcurl4-openssl-dev
dpkg-query: no packages found matching libdb-dev
dpkg-query: no packages found matching libevent-dev
dpkg-query: no packages found matching libffi-dev
dpkg-query: no packages found matching libgdbm-dev
dpkg-query: no packages found matching libgeoip-dev
dpkg-query: no packages found matching libglib2.0-dev
dpkg-query: no packages found matching libjpeg-dev
dpkg-query: no packages found matching libkrb5-dev
dpkg-query: no packages found matching liblzma-dev
dpkg-query: no packages found matching libmagickcore-dev
dpkg-query: no packages found matching libmagickwand-dev
dpkg-query: no packages found matching libncurses5-dev
dpkg-query: no packages found matching libncursesw5-dev
dpkg-query: no packages found matching libpng-dev
dpkg-query: no packages found matching libpq-dev
dpkg-query: no packages found matching libreadline-dev
dpkg-query: no packages found matching libsqlite3-dev
dpkg-query: no packages found matching libssl-dev
dpkg-query: no packages found matching libwebp-dev
dpkg-query: no packages found matching libxml2-dev
dpkg-query: no packages found matching libxslt-dev
dpkg-query: no packages found matching libyaml-dev
dpkg-query: no packages found matching zlib1g-dev
```

- [node:10](https://github.com/nodejs/docker-node/blob/master/10/stretch/Dockerfile) definitely needed, but with `UID=2000` for user `node`
- `npm install moxygen -g` per `moxygen guide`